### PR TITLE
WT-11407 Fix test_txn24 test (not WiredTiger) to stop WT_ROLLBACK errors on MacOS

### PR DESCRIPTION
**Pull request by jeremythorp**
In test_txn24.py, set rollbacks_allowed to 0, and wrapped the final data insertion in a try-except block to catch any rollback errors and keep going.

### CODE REVIEW

File: test/suite/test_txn24.py [hunk 1]
```diff
@@ -112,11 +114,15 @@ def test_snapshot_isolation_and_eviction(self):
...
-            start_row += 1
+
+        try:
+            start_row = 2
+            for i in range(0, n_rows // 4):
+                with self.transaction(session=session4):
+                    cursor4[start_row] = new_val
+                start_row += 1
+        except wiredtiger.WiredTigerRollbackError as e:
+            self.pr("WiredTigerRollbackError occurred while inserting data using cursor4: {}".format(e))
```
Comment (kommiharibabu) [inline test/suite/test_txn24.py:125]: To clarify my understanding, whenever an error occurs, we skip the further insert operation to let the test to pass.

**Comment by Siddhartha8899** [comment]
I believe the main objective of this test is to ensure that WT_ROLLBACK is not returned from the last insertion call, as changes were made to allow the eviction thread to evict the latest committed data. However, we are facing an issue where, due to cache pressure, we encounter ROLLback from the application thread attempting to evict a page from the queue.

The positive aspect is that we are aware of this issue, and even though it also returns a WT_ROLLBACK, we are confident that the original intention behind creating the test still holds true.

Nevertheless, instead of simply skipping the test every time we encounter a WT_ROLLBACK, which would fix the test but make it useless as we would be bypassing the important part of the test, I propose that we explicitly mark the test as skipped. This way, we acknowledge that there is a known issue, there is a ticket to fix it, and the test has been intentionally skipped until the resolution.

By explicitly skipping the test, we avoid any ambiguity for other developers who might come across this test code in the future.

File: test/suite/test_txn24.py [hunk 2]
```diff
@@ -44,6 +44,8 @@ class test_txn24(wttest.WiredTigerTestCase):
     ]
     scenarios = make_scenarios(table_params_values)
 
+    rollbacks_allowed = 0
```
Comment (lukech) [inline test/suite/test_txn24.py:47]: A comment would be helpful to indicate the reason for this setting (encouraging rollback error to occur and recorded in results.txt I believe?)

File: test/suite/test_txn24.py (hunk 1 above)
Comment (jeremythorp) [reply test/suite/test_txn24.py:125]: HI @kommiharibabu , you're correct, that's what this code change would do.

**Comment by jeremythorp** [comment]
Hi @Siddhartha8899 , thanks for your comments and we should consider doing as you suggest. 
If we go down that path, I think we'd do two things: include the `rollbacks_allowed = 0` change, and disable the test on Mac.
What does everyone else think?

File: test/suite/test_txn24.py [hunk 3]
```diff
@@ -52,6 +58,9 @@ def conn_config(self):
                 eviction=(threads_max=4)'
 
     def test_snapshot_isolation_and_eviction(self):
+        if sys.platform.startswith('darwin'):
+            # Skip this test on MacOS/Darwin as it causes WT_ROLLBACKs to occur frequently, and they generate errors.
```
Comment (Siddhartha8899) [inline test/suite/test_txn24.py:62]: I think we should prepend `FIXME-WT-9575` in the comment.

File: test/suite/test_txn24.py (hunk 2 above)
Comment (jeremythorp) [reply test/suite/test_txn24.py:51]: Hi @lukech , I've added a comment explaining why rollbacks_allowed is now set to 0.

File: test/suite/test_txn24.py (hunk 3 above)
Comment (jeremythorp) [reply test/suite/test_txn24.py:63]: Done!

**Comment by jeremythorp** [comment]
I've changed to code to skip this test on MacOS/Darwin, and added a comment with FIXME-WT-9575.

**Review by Siddhartha8899 — APPROVED** [review]
Thanks @jeremythorp. LGTM!

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by lukech — APPROVED** [review]
LGTM

**Comment by jeremythorp** [comment]
Thanks everyone, I'll get the PR merged!

**Pull request merged**